### PR TITLE
07-MemoryManagment: Add module for memory managment task

### PR DESCRIPTION
**Pull request by BekirovBekir**
The module allows to determine required time to memory allocate and free in a kernel

### CODE REVIEW

File: 07-MemoryManagment/memory_managment_module.c [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+	char *kmalloc_ptr;
+	unsigned long long time_ns;
+	size_t order = 0;
+
+	sprintf(ptr_data->buf_str,
+	"size:\t\t\t\t\t\t\t\tallocation time, ns:\t\t\t\t\t\tfreeing time, ns:\n");
+
+		for (order = 0; order < 32; order++) {
+			getnstimeofday(&ptr_data->prev_time);
+			kmalloc_ptr = kmalloc(2 << order, GFP_ATOMIC);
```
Comment (yekovalyov) [inline 07-MemoryManagment/memory_managment_module.c]: Please consider using other flags like GFP_KERNEL for kmalloc.

**Comment by yekovalyov** [comment]
Designed kernel module works as expected but contains a lot of copy-paste code and style can be improved after running checkpatch.pl

**Comment by BekirovBekir** [comment]
GFP_KERNEL flag is used instead GFP_ATOMIC. Output format into sprintf function has been changed. The sprint-functions are used escape sequences.

File: 07-MemoryManagment/memory_managment_module.c [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+void kzalloc_time_calc(struct memory_module_data *ptr_data)
+{
+	char *kzalloc_ptr;
+	unsigned long long time_ns;
+	size_t order = 0;
+
+	sprintf(ptr_data->buf_str,
+		"size:\r\x1b[24Callocation time, ns:\r\x1b[48Cfreeing time, ns:\n");
+
+		for (order = 0; order < 32; order++) {
```
Comment (yekovalyov) [inline 07-MemoryManagment/memory_managment_module.c:116]: It looks like wrong indentation.

File: 07-MemoryManagment/memory_managment_module.c [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+	time_buf = (unsigned long long)(cur->tv_sec) * NS +
+						cur->tv_nsec;
+
+	time_buf -= ((unsigned long long)(prev->tv_sec) * NS +
+						prev->tv_nsec);
+
+	cur->tv_sec = (long)(div_u64_rem(time_buf,
+						NS, &time_remainder));
+
+	cur->tv_nsec = time_remainder;
```
Comment (yekovalyov) [inline 07-MemoryManagment/memory_managment_module.c]: This conversation required multiplication and integer division. It is possible to implement the same conversation using only addition and subtract operations.

File: 07-MemoryManagment/memory_managment_module.c (hunk 2 above)
Comment (BekirovBekir) [reply 07-MemoryManagment/memory_managment_module.c:116]: The checkpatch have found neither error or warning  here.

File: 07-MemoryManagment/memory_managment_module.c (hunk 3 above)
Comment (BekirovBekir) [reply 07-MemoryManagment/memory_managment_module.c]: I've deleted this divison. Please see my latest update.
In my opinion multiplication doesnt take a lot of time. On a modern processor it takes a few clock cycles.

**Pull request merged**